src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -7,61 +7,70 @@
 
 Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
 """
-struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
-  gamma1          ::RealT 
-  gamma2          ::RealT
-  gas_constant1   ::RealT
-  gas_constant2   ::RealT
-  cv1             ::RealT
-  cv2             ::RealT
-  cp1             ::RealT
-  cp2             ::RealT
+
+struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
+  gamma             ::SVector{NCOMP, RealT} 
+  gas_constant      ::SVector{NCOMP, RealT}
+  cv                ::SVector{NCOMP, RealT}
+  cp                ::SVector{NCOMP, RealT}
+  CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT}(gamma, gas_constant) where {NCOMP, NVARS, RealT<:Real} = new(gamma, gas_constant, gas_constant./(gamma.-1), gas_constant + gas_constant./(gamma .-1))
+end
+
+
+function CompressibleEulerMulticomponentEquations2D(; gamma, gas_constant)
+
+  length(gamma) == length(gas_constant) || throw(DimensionMismatch("gamma and gas_constant should have the same length"))
+  NVARS = length(gamma) + 3
+  NCOMP = length(gamma)
+
+  return CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, Float64}(gamma, gas_constant)
 end
 
-function CompressibleEulerMulticomponentEquations2D()
 
-  # Set ratio of specific heat of the gas per species 
-  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
-  gamma1  = 1.4
-  gamma2  = 1.4
+get_name(::CompressibleEulerMulticomponentEquations2D) = "CompressibleEulerMulticomponentEquations2D"
+
+
+function varnames(::typeof(cons2cons), equations::CompressibleEulerMulticomponentEquations2D)
+  @unpack gamma = equations
+
+  NCOMP = length(gamma)
+  NVARS = length(gamma) + 3
   
```

Comment:
This part is deliberately type-unstable. Although it's not performance-critical, it might be nice to use a type-stable version, e.g. something like
```suggestion
function varnames(::typeof(cons2cons), equations::CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS}) where {NCOMP, NVARS}

  cons = ("rho_v1", "rho_v2", "rho_e")
  rhos = ntuple(n -> "rho" * string(n), Val(NCOMP))
  return SVector{NVARS}(cons..., rhos...)
end
```
(+ deleting the other lines changed in this function below). The gist of this is "If you want a tuple (or `SVector`, ...) in a type-stable way, do not use plain `Array`s for intermediate values when creating it".

Reply:
done in 4c720b0